DCGM builds for multiple CUDA versions, but our CUDA packages aren't
coinstallable.  To work around this, we change the CUDA discovery mechanism to
point at CUDA 13 for 11 and 12 and then remove the "incorrectly" built
artifacts afterwards.

Add /usr/local as a valid path for CUDA, and do not use arch-specific target
directory.

This also makes it possible to specify CUDA versions during configuration with
cmake.
--- a/cmake/FindCuda.cmake
+++ b/cmake/FindCuda.cmake
@@ -14,7 +14,7 @@
 # limitations under the License.
 #
 
-set(_Cuda_VERSIONS 11.8 12.9 13.0)
+set(CMAKE_CUDA_VERSIONS "" CACHE STRING "CUDA versions to build against.")
 
 if (CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
     set(_Cuda_TARGET_DIR "targets/x86_64-linux")
@@ -46,11 +46,13 @@ foreach (_Cuda_PATH IN ITEMS Cuda_ROOT CUDA_ROOT)
 endforeach()
 
 unset(_Cuda_PATH)
+list(APPEND _Cuda_PATHS "/usr/local")
+set(_Cuda_TARGET_DIR "")
 
 list(SORT _Cuda_PATHS)
 list(REMOVE_DUPLICATES _Cuda_PATHS)
 
-foreach(_Cuda_VERSION IN LISTS _Cuda_VERSIONS)
+foreach(_Cuda_VERSION IN LISTS CMAKE_CUDA_VERSIONS)
     set(_Cuda_FOUND TRUE)
 
     string(REGEX MATCH "^[^.]+" _Cuda_MAJOR_VERSION "${_Cuda_VERSION}")
@@ -152,6 +154,16 @@ foreach(_Cuda_VERSION IN LISTS _Cuda_VERSIONS)
         "${Cuda_${_Cuda_MAJOR_VERSION}_cublasLt_LIBRARY}")
 endforeach()
 
+set(CUDA11_INCLUDE_DIR "${CUDA13_INCLUDE_DIR}")
+set(CUDA11_STATIC_LIBS "${CUDA13_STATIC_LIBS}")
+set(CUDA11_LIBS "${CUDA13_LIBS}")
+set(CUDA11_STATIC_CUBLAS_LIBS "${CUDA13_STATIC_CUBLAS_LIBS}")
+
+set(CUDA12_INCLUDE_DIR "${CUDA13_INCLUDE_DIR}")
+set(CUDA12_STATIC_LIBS "${CUDA13_STATIC_LIBS}")
+set(CUDA12_LIBS "${CUDA13_LIBS}")
+set(CUDA12_STATIC_CUBLAS_LIBS "${CUDA13_STATIC_CUBLAS_LIBS}")
+
 unset(_Cuda_LIBRARY)
 unset(_Cuda_FOUND)
 unset(_Cuda_VERSION)
